### ISSUE TITLE
Support for Nunchuck and Accelerometer!

### ISSUE THREAD

**Issue by VIPTankz** [Felk/dolphin#41]
Trying to get some AIs to play Super Mario Galaxy and some other games so was hoping I could use the nunchuck and accelerometer for shaking!

**Comment by VIPTankz** [comment]
Thanks for the comment, however I'm aware that dolphin has this feature, but I need to be able to do this through the python interface implemented by this fork of dolphin, not using key presses. The reason for this is that I often run many (sometimes up to 20) identical instances of dolphin at once, so if I used key presses they would interrupt each other. Plus, it would render that key unusable on my PC for the time the AI is training.

What I'm hoping someone can do is use the feature you mention to add a python stub, similar to the others that are already implemented in dolphin/python-stubs/controller.pyi

**Comment by PlanetSat0rn** [comment]
i expected this issue page to be large judging from the views on your video but guess i was wrong

**Comment by Felk** [comment]
I tried adding support for Nunchuks and the accelerometer and finally documented it in 16036d7628249aaf67167f6ec04972f83a7d825b, please try it out! Also let me know if you can't run from master need a precompiled build instead

**Comment by VIPTankz** [comment]
Thank you very very much, I'll be trying it out very soon! If you could release a precompiled build I would be extremely grateful! I saw the messages before about #42, I think adding point, shake and tilt would make things simpler for many games.

**Comment by Felk** [comment]
I made another prerelease that contains a binary for windows: https://github.com/Felk/dolphin/releases/tag/scripting-preview3

> I think adding point, shake and tilt would make things simpler for many games.

I'll let this issue open for now, to track that

**Comment by VIPTankz** [comment]
Just been testing the nunchuck and think I've found an issue. I'm currently trying to play Super Mario Galaxy, but regardless of which way I set the nunchuck stick, Mario will only move Up/Right, depending on if I'm editing StickX or StickY. I've tried using 1 and -1, but both ways seem to cause motion in the same direction. I've also tried using 0.5 and -0.5, but has the same result.

I will include the code I've used to recreate the issue, however this includes a savestate from the game Super Mario Galaxy. You may be able to recreate the issue in other games that use the nunchuck though.

```
from dolphin import event, gui, savestate, memory, controller
import time
red = 0xffff0000

wii_dic = {"A": False,
                "B": False,
                "One": False,
                "Two": False,
                "Plus": False,
                "Minus": False,
                "Home": False,
                "Up": False,
                "Down": False,
                "Left": False,
                "Right": False}
nun_dic = {"C": False,
                "Z": False,
                "StickX": 0,
                "StickY": 0}

steps = 0
start = time.time()

controller.set_wii_nunchuk_buttons(0, nun_dic)
controller.set_wiimote_buttons(0, wii_dic)
for i in range(4):
    (width, height, data) = await event.framedrawn()

savestate.load_from_slot(1)

while True:

    (width, height, data) = await event.framedrawn()

    steps += 1
    fps = round(steps / (time.time() - start))
    gui.draw_text((10, 10), red, "AVG FPS: " + str(fps))

    #Moves right
    #nun_dic["StickX"] = 1

    #should move left, but moves right?
    nun_dic["StickY"] = -1

    controller.set_wii_nunchuk_buttons(0, nun_dic)
    controller.set_wiimote_buttons(0, wii_dic)
```

**Comment by Felk** [comment]
Oops, yeah you found a copy paste mistake. I fixed it in 6e34533b7572431b84cf1143463f01c8d9777cd2. Turns out I accidentally treated them as digital buttons instead of analog inputs, causing everything not being equal to "0" be turned into "1".

[dolphin-x64-exe-scripting-preview3-fix-nunchuk-stick.zip](https://github.com/Felk/dolphin/files/12732257/dolphin-x64-exe-scripting-preview3-fix-nunchuk-stick.zip)
